torch/testing/_internal/common_optimizers.py

```diff
@@ -99,17 +99,24 @@ def __init__(
         self.error_regex = error_regex
 
 
+# Typing for every optim_inputs_func
+class OptimInputsFunc:
```

Comment:
There's the shorter one line way of defining function types in python like https://github.com/pytorch/pytorch/blob/efab8c433f257a4ab01fd07ca4b4a6aaf6ddc632/torch/optim/optimizer.py#L47

How does this `def __call__...` compare?